elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -75,18 +92,15 @@ public ElasticsearchSchemaFactory() {
 
   /**
    * Builds elastic rest client from user configuration
-   * @param coordinates list of {@code hostname/port} to connect to
+   * @param hostSet set of ES HTTP Hosts to connect to
    * @return newly initialized low-level rest http client for ES
    */
-  private static RestClient connect(Map<String, Integer> coordinates) {
-    Objects.requireNonNull(coordinates, "coordinates");
-    Preconditions.checkArgument(!coordinates.isEmpty(), "no ES coordinates specified");
-    final Set<HttpHost> set = new LinkedHashSet<>();
-    for (Map.Entry<String, Integer> entry: coordinates.entrySet()) {
-      set.add(new HttpHost(entry.getKey(), entry.getValue()));
-    }
+  private static RestClient connect(Set<HttpHost> hostSet) throws MalformedURLException {
```

Comment:
Make hosts `Collection` or `List` (ES client will do the de-dupe)

Reply:
Done.